blackwidowlib/src/main/java/io/github/arcaneplugins/blackwidow/lib/cmdblocking/UpdateChecker.java

```diff
@@ -0,0 +1,58 @@
+package io.github.arcaneplugins.blackwidow.lib.cmdblocking;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Scanner;
+import java.util.function.Consumer;
+
+/**
+ * @author stumper66
+ * @since 1.1.0
+ */
+public class UpdateChecker {
+
+    public UpdateChecker(
+            final String resourceName
+    ) {
+        this.resourceName = resourceName;
+    }
+
+    private final String resourceName;
+    private String errorMessage = null;
+
+    public boolean success() {
+        return errorMessage != null;
+    }
+
+    @Nullable
+    public String errorMessage() {
+        return errorMessage;
+    }
+
+    // returns if successful or not
+    public boolean checkLatestVersion(
```

Comment:
I'll convert this over to an exception system rather than booleans + error message variables - might be cleaner to handle errors, or otherwise easier to backtrace

Reply:
should be resolved in latest commit - will look over later to confirm nothing else to do on this